src/main/scala/by/itechart/tutorial/web/GroupsApiRouter.scala

```diff
@@ -0,0 +1,53 @@
+package by.itechart.tutorial.web
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{complete, concat, get, onSuccess, pathPrefix, _}
+import akka.http.scaladsl.server.Route
+import by.itechart.tutorial.dao.{Group, GroupRepository}
+import by.itechart.tutorial.service.GroupService
+import by.itechart.tutorial.util.const.Constants.MODEL_NOT_FOUND
+import javax.inject.Inject
+
+class GroupsApiRouter @Inject()(groupService: GroupService) {
+
+  def getGroupsApiRoutes: Route = route
+
+  import by.itechart.tutorial.util.marshalling.GroupJsonProtocol._
+
+  private val route: Route =
+    concat(
+      get {
+        pathPrefix("groups" / LongNumber) { id =>
+          val maybeItem = groupService.getGroupById(id)
+          onSuccess(maybeItem) {
+            case Some(group) => complete(group)
+            case None => complete(MODEL_NOT_FOUND.format("group"))
```

Comment:
This will still produce application/text response, which I hard to use on client side. Make sure you always return some JSON, even if it will have one field "message"